### ISSUE TITLE
Import fails: JSON.parse: unexpected character

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/contacts#880]
When trying to import a Contacts.vcf I just exported from my ownCloud instance, I get this yellow notification error:
`SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data: parsererror`

In the JS console:
```
"Import clicked" addressbooks.js:379:3
Object { readyState: 4, getResponseHeader: .ajax/C.getResponseHeader(), getAllResponseHeaders: .ajax/C.getAllResponseHeaders(), setRequestHeader: .ajax/C.setRequestHeader(), overrideMimeType: .ajax/C.overrideMimeType(), statusCode: .ajax/C.statusCode(), abort: .ajax/C.abort(), state: .Deferred/r.state(), always: .Deferred/r.always(), then: .Deferred/r.then(), 12 de plus… } storage.js:707:4
"jqXHR" Object { readyState: 4, getResponseHeader: .ajax/C.getResponseHeader(), getAllResponseHeaders: .ajax/C.getAllResponseHeaders(), setRequestHeader: .ajax/C.setRequestHeader(), overrideMimeType: .ajax/C.overrideMimeType(), statusCode: .ajax/C.statusCode(), abort: .ajax/C.abort(), state: .Deferred/r.state(), always: .Deferred/r.always(), then: .Deferred/r.then(), 12 de plus… } storage.js:12:2
"fail" Object { disabled: false, create: null, dropZone: Object, replaceFileInput: true, singleFileUploads: true, limitMultiFileUploadSizeOverhead: 512, sequentialUploads: false, forceIframeTransport: false, multipart: true, recalculateProgress: true, 46 de plus… }
```

And this in the owncloud.log
```
{"reqId":"ATvfOgsJxMq6L+3KZRJO","remoteAddr":"127.0.0.1","app":"no app in context",
"message":"Accessing the API class is deprecated! Use the appropriate services instead!",
"level":0,"time":"2015-04-15T20:20:42+00:00","method":"GET",
"url":"\/owncloud\/index.php\/apps\/contacts\/addressbooks\/"}
{"reqId":"jyp57Ts+Be+S1lEiMAAU","remoteAddr":"127.0.0.1","app":"no app in context",
"message":"Accessing the API class is deprecated! Use the appropriate services instead!",
"level":0,"time":"2015-04-15T20:20:48+00:00","method":"POST",
"url":"\/owncloud\/index.php\/apps\/contacts\/addressbook\/local\/1\/automatic\/import\/upload"}
{"reqId":"jyp57Ts+Be+S1lEiMAAU","remoteAddr":"127.0.0.1","app":"PHP",
"message":"Class 'OC_FileProxy' not found at \/home\/jan\/owncloud\/apps\/contacts\/lib\/controller\/importcontroller.php#89",
"level":3,"time":"2015-04-15T20:20:48+00:00","method":"POST",
"url":"\/owncloud\/index.php\/apps\/contacts\/addressbook\/local\/1\/automatic\/import\/upload"}
```

@jbtbnl @babelouest @LEDfan any idea?

**Comment by babelouest** [comment]
I don't know if it's related, but I have a problem importing VCard 3.0 files too:
`PHP Fatal error:  Call to a member function getDocumentType() on a non-object in [...]/core/3rdparty/sabre/vobject/lib/Parameter.php on line 287`

If I comment out those 3 lines in the Parameter.php file, the import works again:
```PHP
        if ($this->root->getDocumentType() === Document::VCARD21 && $this->noName) {

            return implode(';', $value);

        }
```

**Comment by babelouest** [comment]
Follow-up for the issue I was talking about.

Instant messaging properties are `X-` properties in gmail and are translated into `IMPP` properties in owncloud import. Import also adds an `X-SERVICE-TYPE` property.

In Sabre library, an `X-SERVICE-TYPE` node has no method called `getDocumentType`, so the error message. If i replace line 287 by a simple check, it works too:
```PHP
if (method_exists($this->root, "getDocumentType") && $this->root->getDocumentType() === Document::VCARD21 && $this->noName) {
```

**Comment by jancborchardt** [comment]
@babelouest so can you submit a pull request as fix? :) Or how do we proceed?

Also cc @DeepDiver1975

**Comment by babelouest** [comment]
Here it is: https://github.com/owncloud/3rdparty/pull/172
although that's just a workaround, I couldn't find why in some cases the object `$this->root` in `Parameter.php` is a string and not a Node or a Document.

**Comment by evert** [comment]
Hey guys... the proposed fix in sabre/vobject does not make a lot of sense. In any case it's fixing a symptom of a bug, but not the underlying cause.

I suspect it's an owncloud bug, but I could be wrong. Could you try to figure out the root cause for this? Even if it's not a vobject bug, I'm interested to find out what situation could have triggered this, so we can add a harder check in vobject to prevent this from ever happening again.

**Comment by babelouest** [comment]
I think I found the underlying bug, it was due to the creation of a VObject Parameter here https://github.com/owncloud/contacts/blob/master/lib/connector/importconnector.php#L69
there was no $root parameter passed.

I made a fix that works on my config (#903 ), can you check if that fix your problems too ?
